bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/internal/YamlModelRepositoryImpl.java

```diff
@@ -99,7 +99,8 @@ public YamlModelRepositoryImpl(@Reference(target = WatchService.CONFIG_WATCHER_F
 
         // read initial contents
         try (Stream<Path> files = Files.walk(watchPath)) {
-            files.filter(Files::isRegularFile).map(watchPath::relativize).forEach(f -> processWatchEvent(CREATE, f));
+            files.filter(f -> Files.isReadable(f) && Files.isRegularFile(f)).map(watchPath::relativize)
```

Comment:
Is it really necessary to check `.isReadable` here? The check is done again when `processWatchEvent` is called.

Reply:
I was not sure what call triggered the AccessDeniedException. Not really clear when looking at the stack trace.